src/fdosecrets/widgets/SettingsWidgetFdoSecrets.cpp

```diff
@@ -301,4 +312,53 @@ void SettingsWidgetFdoSecrets::saveSettings()
     FdoSecrets::settings()->setNoConfirmDeleteItem(m_ui->noConfirmDeleteItem->isChecked());
 }
 
+void SettingsWidgetFdoSecrets::showMessage(const QString& text, MessageWidget::MessageType type)
+{
+    // Show error messages for a longer time to make sure the user can read them
+    if (type == MessageWidget::Error) {
+        m_ui->warningMsg->setCloseButtonVisible(true);
+        m_ui->warningMsg->showMessage(text, type, -1);
+    } else {
+        m_ui->warningMsg->setCloseButtonVisible(false);
+        m_ui->warningMsg->showMessage(text, type, 2000);
+    }
+}
+
+void SettingsWidgetFdoSecrets::showEvent(QShowEvent* event)
+{
+    QWidget::showEvent(event);
+    QTimer::singleShot(0, this, &SettingsWidgetFdoSecrets::checkDBusName);
+    m_checkTimer.start();
+}
+
+void SettingsWidgetFdoSecrets::hideEvent(QHideEvent* event)
+{
+    QWidget::hideEvent(event);
+    m_checkTimer.stop();
+}
+
+void SettingsWidgetFdoSecrets::checkDBusName()
+{
+    if (m_plugin->serviceInstance()) {
+        // only need checking if the service is not started or failed to start.
+        return;
+    }
+
+    qDebug() << "Checking DBus Name";
+    auto reply = QDBusConnection::sessionBus().interface()->isServiceRegistered(QStringLiteral(DBUS_SERVICE_SECRET));
+    if (!reply.isValid()) {
+        showMessage(tr("Failed to connect to DBus. Please check your DBus setup."), MessageWidget::Error);
+        return;
+    }
+    if (reply.value()) {
+        auto otherService = m_plugin->runningServiceInfo();
+        showMessage(tr("<b>Warning:</b> Another secret service is running:<br/>")
+                        + QStringLiteral("<i>%1</i><br/>").arg(otherService)
+                        + tr("Please stop/remove it before re-enabling the Secret Service Integration."),
+                    MessageWidget::Warning);
```

Comment:
Need to escape this as well. I am also not so happy with duplicating such a long error message in two places. Can we consolidate this somehow?

Reply:
Me either. I thought about the wording, but couldn't come up with something better. They are two different things in the end: one is the fact that the service failed to start, with a reason; this one is a warning. Any suggestions?